src/metrics/scrapper.py

```diff
@@ -1,36 +1,41 @@
 # Python Imports
 import logging
+import socket
 import pandas as pd
 from itertools import chain
 from typing import List, Dict
 from pathlib import Path
+from kubernetes.client import CoreV1Api
+from result import Ok, Err, Result
 
 # Project Imports
 from src.metrics import scrape_utils
-from result import Ok, Err, Result
+from src.metrics import kubernetes
 from src.utils.file_utils import read_yaml_file
-from src.utils.queries import get_query_data
 
 logger = logging.getLogger(__name__)
 
 
 class Scrapper:
-    def __init__(self, url: str, query_config_file: str, out_folder: str):
+    def __init__(self, api: CoreV1Api,  url: str, query_config_file: str, out_folder: str):
         self._url = url
         self._query_config = None
         self._query_config_file = query_config_file
         self._out_folder = out_folder
         self._set_query_config()
-        # TODO make interval match value in cluster
+        self._k8s = kubernetes.KubernetesManager(api)
 
     def query_and_dump_metrics(self):
+        socket.create_connection = self._k8s.create_connection
```

Comment:
I'd suggest leaving a comment so future people (including us) understand why in heavens this is necessary 😆

Reply:
Yup, thought about that as well. Will add it.